pynucastro/rates/rate.py

```diff
@@ -240,9 +244,14 @@ def _set_print_representation(self):
                     rhs_other.append("nubar")
 
         for n, r in enumerate(treactants):
-            self.string += f"{r.c()}"
-            self.rid += f"{r}"
-            self.pretty_string += fr"{r.pretty}"
+            factor = ""
+            if self.stoichiometry:
+                c = self.stoichiometry[r]
```

Comment:
currently we have to set up reactant/product explictly to be 1 if we set up stoichiometry otherwise it gives error, but it feels redundant. I think we should check keys first, then proceed.

Reply:
okay, I think this is done now + I added a test